raiden/smart_contracts/NettingChannelLibrary.sol

```diff
@@ -385,19 +358,36 @@ library NettingChannelLibrary {
         }
 
         self.settled = block.number;
-
         total_netted = node1.netted + node2.netted;
         total_deposit = node1.balance + node2.balance;
 
-        if (total_netted != total_deposit) {
-            throw;
+        Participant memory closing_party;
+        Participant memory other_party;
+
+        if (node1.node_address == self.closing_address) {
+            closing_party = node1;
+            other_party = node2;
+        } else {
+            closing_party = node2;
+            other_party = node1;
         }
 
-        if (!self.token.transfer(node1.node_address, node1.netted)) {
-            throw;
+
+        // first pay out to the party that did not close the channel
+        uint amount = total_deposit < other_party.netted
+            ? total_deposit
+            : other_party.netted;
+        if (amount > 0) {
+            if (!self.token.transfer(other_party.node_address, amount)) {
+                throw;
+            }
         }
-        if (!self.token.transfer(node2.node_address, node2.netted)) {
-            throw;
+        // then payout whatever can be paid out to the closing party
```

Comment:
I'd rather have the closing party only get whatever they could *prove* to the contract.